internal/servicedeployer/factory.go

```diff
@@ -68,8 +87,18 @@ func Factory(options FactoryOptions) (ServiceDeployer, error) {
 		if _, err := os.Stat(customAgentCfgYMLPath); err != nil {
 			return nil, fmt.Errorf("can't find expected file custom-agent.yml: %w", err)
 		}
-		return NewCustomAgentDeployer(options.Profile, customAgentCfgYMLPath, options.StackVersion)
+		opts := CustomAgentDeployerOptions{
+			Profile:           options.Profile,
+			DockerComposeFile: customAgentCfgYMLPath,
+			StackVersion:      options.StackVersion,
+			RunTearDown:       options.RunTearDown,
+			RunTestsOnly:      options.RunTestsOnly,
+		}
+		return NewCustomAgentDeployer(opts)
 	case "tf":
+		if options.RunSetup || options.RunTearDown || options.RunTestsOnly {
+			return nil, errors.New("terraform service deployer not supported to run by steps")
+		}
```

Comment:
For now, this service deployer is not supported. I'd prefer to just leave this option (test flags) for stacks/services started locally for now.

This the error that is going to be shown to the user:

```
2024/02/15 09:44:36 DEBUG setting up service...
Error: error running package system tests: could not complete test run: could not create service runner: terraform service deployer not supported to run by steps
```

Reply:
:+1: we can add this later